cypress/integration/team/create_a_team_spec.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getRandomInt} from '../../utils';
+
+// ***************************************************************
+// - [number] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Teams Suite', () => {
+    before(() => {
+        // 1. Login and go to /
+        cy.login('user-1');
+        cy.visit('/');
+    });
+
+    it('Create a team', async () => {
+        const teamURL = `team-test-${getRandomInt(9999).toString()}`;
+
+        //  Click hamburger main menu
+        cy.get('#sidebarHeaderDropdownButton').click();
+
+        // dropdown menu should be visible then click create team
+        cy.get('#createTeam').should('be.visible').click();
```

Comment:
Above didn't work for me.   It's failing to find `#createTeam`.  I think it should be:

```suggestion
        // * Dropdown menu should be visible
        cy.get('#sidebarDropdownMenu').should('be.visible').within(() => {
            // 3. Click "Create a New Team"
            cy.get('#createTeam').should('be.visible').click();
        });
```

Reply:
using within will failed on code style, because in eslint maximum nested callback is 3. but within need add a callback.
I prefer using this 
`// * Dropdown menu should be visible
        cy.get('#sidebarDropdownMenu').should('be.visible');

        // 3. Click "Create a New Team"
        cy.get('#sidebarDropdownMenu li #createTeam').click();`
or 
`// 3. Click "Create a New Team"
        cy.get('#sidebarDropdownMenu').should('be.visible').find('#createTeam').click();`

what do you think?

Reply:
What you have as first option looks good to me 🎉